deps/src/main/java/com/microsoft/azure/sdk/iot/deps/auth/IotHubSSLContext.java

```diff
@@ -196,7 +196,7 @@ private void generateSSLContextWithKeys(String publicKeyCertificateString, Strin
         Key privateKey = IotHubSSLContext.parsePrivateKey(privateKeyString);
         Collection<X509Certificate> certChain = IotHubSSLContext.parsePublicKeyCertificate(publicKeyCertificateString);
 
-        X509Certificate[] certs = certChain.toArray(new X509Certificate[certChain.size()]);
+        X509Certificate[] certs = certChain.toArray(new X509Certificate[0]);
```

Comment:
is this always 0?

Reply:
The newly created array will not be size 0, but for the sake of converting a collection to an array, this is the recommended way, yes

Reply:
"There are two styles to convert a collection to an array: either using a pre-sized array (like c.toArray(new String[c.size()])) or using an empty array (like c.toArray(new String[0]).
In older Java versions using pre-sized array was recommended, as the reflection call which is necessary to create an array of proper size was quite slow. However since late updates of OpenJDK 6 this call was intrinsified, making the performance of the empty array version the same and sometimes even better, compared to the pre-sized version. Also passing pre-sized array is dangerous for a concurrent or synchronized collection as a data race is possible between the size and toArray call which may result in extra nulls at the end of the array, if the collection was concurrently shrunk during the operation.
This inspection allows to follow the uniform style: either using an empty array (which is recommended in modern Java) or using a pre-sized array (which might be faster in older Java versions or non-HotSpot based JVMs)."